compiler/src/lnntoamm/Expr.ts

```diff
@@ -607,7 +654,9 @@ class Call extends Expr {
       console.log('expected output type:', ty);
       throw new Error(`no function selected`);
     }
-    const fn = selected.pop();
+    // Fn.select should implement the matrix so that the most reasonable
+    // choice is last in the fn array.
+    const fn = selFns.pop();
```

Comment:
Perhaps a warning mode might be worthwhile to allow users to know that the type was ambiguous and it's choosing from a list? "most reasonable" might not be what they're actually after and letting them know they should annotate their desired type seems like a good idea?

Reply:
that's a good idea, but that'd have to wait for a future PR that focuses on better error handling throughout the compiler in general. the "most reasonable" wording has 2 meanings here: first, it prefers functions that are most specialized for the given argument types (for example, it picks `fn add(int64, int64)` over `fn add(int32, int32)` given raw literals, even if the latter was declared after the former since we prefer `int64` over `int32` in alan - this really only means that it impacts such cases). The other meaning is that it still maintains order of definition, so the last-defined function is given higher precedence over the first-defined function, even if both have the same function signature.

The comment is slightly outdated (I made it in one of the earlier commits before I moved the matrix selection to `Types.ts`) so I'll update it and add clarity

Reply:
Yeah, no problem, just pointing out that this is a good place for eventually adding a warning that the compiler could emit and also maybe a fail-on-warning mode check if we want that, too.